src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp

```diff
@@ -2058,6 +2058,9 @@ int AppleseedRenderer::Render(
     if (!m_rend_params.inMtlEdit || m_settings.m_log_material_editor_messages)
         create_log_window();
 
+    if (renderer_settings.m_noise_seed == 0)
+        renderer_settings.m_noise_seed = static_cast<int>(time) / GetTicksPerFrame();
```

Comment:
Not sure the division makes sense. Noise seed should vary for every frame, not just every second.

Reply:
That is exactly what it does. In Max time is represented in ticks and a frame duration can vary in time depending on the frames/second setting (24,25 or 30). The GetTicksPerFrame() call returns the duration of the current frame in ticks.  By dividing the accumulated time (in ticks) by the frame duration (in ticks) I get the current frame number which varies the noise seed.
`renderer_settings.m_noise_seed += static_cast<int>(time) / GetTicksPerFrame();`

Reply:
Got it, thanks for the explanation! I was thrown off by `GetTicksPerFrame()`, I somehow read it as `GetTicksPerSecond()` or something.

May I suggest introducing a temporary variable `frame_number` to clarify that logic?

Reply:
ok, I added a temporary variable.